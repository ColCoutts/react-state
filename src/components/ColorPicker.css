.ColorPicker {
  height: 100vh;

  button {
    cursor: pointer;
    width: 20%;
    height: 25%;
  }

  div {
    width: 15%;
    height: 15%;
  }

  .red {
    background-color: red;
  }

  .blue {
    background-color: blue;
  }

  .green {
    background-color: green;
  }

}
