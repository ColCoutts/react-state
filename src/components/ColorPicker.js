import React from 'react';
import PropTypes from 'prop-types';
import styles from './ColorPicker.css';

function ColorPicker({ selectColorHandler }) {
  return (
    <div className={styles.ColorPicker}>
      <button className={styles.red} onClick={() => this.selectColorHandler()}>RED</button>
      <button className={styles.green} onClick={() => this.selectColorHandler()}>GREEN</button>
      <button className={styles.blue} onClick={() => this.selectColorHandler()}>BLUE</button>
    </div>
  );
}

ColorPicker.propTypes = {
  selectColorHandler: PropTypes.func.isRequired
};

export default ColorPicker;
