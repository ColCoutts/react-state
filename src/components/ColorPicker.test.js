import React from 'react';
import { shallow } from 'enzyme';
import ColorPicker from './ColorPicker';

describe('ColorPicker component', () => {
  it('renders a ColorPicker', () => {
    const wrapper = shallow(<ColorPicker selectColorHandler={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('selects a red button on click', () => {
    const selectColorHandler = jest.fn();
    const wrapper = shallow(<ColorPicker selectColorHandler={selectColorHandler} />);

    wrapper.find('button').at(0).simulate('click');

    expect(selectColorHandler).toHaveBeenCalledWith('red');
  });

});
