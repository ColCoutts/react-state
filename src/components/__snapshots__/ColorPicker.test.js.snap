// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders a ColorPicker 1`] = `
<section
  className="ColorPicker"
>
  <button
    className="red"
    onClick={[Function]}
  >
    RED
  </button>
  <button
    className="green"
    onClick={[Function]}
  >
    GREEN
  </button>
  <button
    className="blue"
    onClick={[Function]}
  >
    BLUE
  </button>
</section>
`;
